pkg/controllers/generic/import_syncer.go

```diff
@@ -72,23 +83,36 @@ func CreateImporters(ctx *context2.ControllerContext, config *config.Config) err
 	return nil
 }
 
-func createImporter(ctx *synccontext.RegisterContext, config *config.Import) (syncer.Syncer, error) {
-	statusIsSubresource := true
-	// TODO: [low priority] check if config.Kind + config.APIVersion has status subresource
-
+func createImporter(ctx *synccontext.RegisterContext, config *config.Import, gvkRegister GVKRegister) (syncer.Syncer, error) {
 	gvk := schema.FromAPIVersionAndKind(config.APIVersion, config.Kind)
-	controllerID := fmt.Sprintf("%s/%s/GenericImport", strings.ToLower(gvk.Kind), strings.ToLower(gvk.Group))
+	controllerID := fmt.Sprintf("%s/%s/GenericImport", strings.ToLower(gvk.Kind), strings.ToLower(gvk.GroupVersion().String()))
+
+	syncerOptions := &syncer.Options{
+		DisableUIDDeletion: true,
+	}
+
+	if scopeAndSubresource, ok := gvkRegister[gvk]; ok {
+		if scopeAndSubresource.IsClusterScoped {
+			syncerOptions.IsClusterScopedCRD = true
+		}
+
+		if scopeAndSubresource.HasStatusSubresource {
+			syncerOptions.HasStatusSubresource = true
+		}
```

Comment:
Could this be simplified or am I missing something?
```suggestion
		syncerOptions.IsClusterScopedCRD = scopeAndSubresource.IsClusterScoped
		syncerOptions.HasStatusSubresource = scopeAndSubresource.HasStatusSubresource
```